renku_notebooks/api/schemas/cloud_storage.py

```diff
@@ -15,21 +17,43 @@ class Meta:
     bucket = fields.Str(required=True, validate=validate.Length(min=1))
 
     @post_load
-    def create_s3mount_object(self, data, **kwargs):
+    def create_cloud_storage_object(self, data, **kwargs):
         if data["access_key"] == "":
             data.pop("access_key")
         if data["secret_key"] == "":
             data.pop("secret_key")
-        s3mount = S3mount(**data, mount_folder="/cloudstorage", read_only=True)
-        if not s3mount.bucket_exists:
+
+        if (
+            data.get("access_key") is None
+            and data.get("secret_key") is not None
+            and config.cloud_storage.azure_blob.enabled
+        ):
+            cloud_storage = AzureBlobRequest(
+                endpoint=data["endpoint"],
+                container=data["bucket"],
+                credential=data["secret_key"],
+                mount_folder=config.cloud_storage.mount_folder,
+                read_only=config.cloud_storage.azure_blob.read_only,
+            )
+        else:
```

Comment:
Shouldn't we check that s3 is enabled here via `config.cloud_storage.s3.enabled`?

Also we should fail is both s3.enabled=True and azure_blob.enabled=True in the values file somewhere (I didn't see code for this)

Reply:
@Panaetius for now there is no problem if both s3 and azure blob storage are enabled. And I foresee that some deployments will want to use both. Especially those running on Azure. But even non-Azure deployments can use both s3 and azure provided that they have installed the azure blob csi drivers in their k8s cluster.

The way I recognize azure blob storage is if the access key is not provided by the secret key is provided. That is the only way. It is not perfect but it should work to start. Then we can adapt the ui and have proper indication what storage type is being used. But for now we are jumping onto the same api for the s3 storage. 

When the user mounts a azure blob container then the endpoint is still the azure endpoint, the bucket is the azure blob container name and the secret key is the azure blob credentials. And the public key is left blank.

So I will add the checks for what is enabled and fail if s3 stuff is provided but s3 is disabled. But we can definitely support both azure blob and s3.